kube-aws-updater

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+# Shell style guide: https://google.github.io/styleguide/shell.xml
+
+# ## (-) master and worker steps
+# ## (*) worker specific steps
+#
+# - label nodes
+# - disable scheduling
+# - double ASG size
+# - wait for 2x group
+# - stop ASG actions
+#   * delete docker-reigistry pod
+#   * wait for docker-registry to be up
+# - drain/terminate labelled nodes (sequentially)
+# - resize ASG to original size
+# - re-enable ASG actions
+
+set -e
+
+retire_time=$(date +"%Y-%m-%dT%H-%M-%SZ")
+
+# flags
+kube_context=''
+aws_profile=''
+role=''
+registry_flag=''
+
+usage() {
+  cat <<EOF
+Usage: $0 -c <kube_context> -p <aws_profile> -r [role] [-e]
+
+  -e    Registry flag - set if cluster contains docker registry and
+        needs to be moved prior to rest of the pods"
+EOF
+}
+
+while getopts 'c:p:r:eh' flag; do
+  case "${flag}" in
+    c) kube_context="${OPTARG}" ;;
+    p) aws_profile="${OPTARG}" ;;
+    r) role="${OPTARG}" ;;
+    e) registry_flag='true' ;;
+    h) usage && exit 0 ;;
+    *) echo "Unexpected option ${flag}" && usage && exit 1 ;;
+  esac
+done
+
+### Deps
+deps="
+  jq
+  kubectl
+  aws
+"
+
+missing_deps=
```

Comment:
You might want to change my code to explicitly initialize variables as stings using quotes.

Reply:
Resolved